test/src/tests.js

```diff
@@ -139,4 +174,577 @@ describe('Bing Ads Event Forwarder', function() {
+            ];
+
+            window.uetq.length.should.eql(3);
+            window.uetq[0].should.equal('consent');
+            window.uetq[1].should.equal('default');
+            window.uetq[2].should.eql(expectedInitialConsentPayload[2]);
+
+            var obj = {
+                EventDataType: MessageType.PageEvent,
+                EventName: 'Test Page Event',
+                CustomFlags: {
+                    'Bing.EventValue': 10,
+                },
+                ConsentState: {
+                    getGDPRConsentState: function() {
+                        return {
+                            marketing_consent: {
+                                Consented: true,
+                                Timestamp: 1557935884509,
+                                ConsentDocument: 'Marketing_Consent',
+                                Location: 'This is fake',
+                                HardwareId: '123456',
+                            },
+                        };
+                    },
+
+                    getCCPAConsentState: function() {
+                        return {
+                            data_sale_opt_out: {
+                                Consented: false,
+                                Timestamp: Date.now(),
+                                Document: 'some_consent',
+                            },
+                        };
+                    },
+                },
+            };
+
+            mParticle.forwarder.process(obj);
+
+            // UETQ should now have 7 elements
+            // The first 3 elements of this array will be the consent payload
+            // The next 3 elments will be the consent update payload
+            // The 7th element will be the event payload
+
+            window.uetq.length.should.eql(7);
+            window.uetq[3].should.equal('consent');
+            window.uetq[4].should.equal('update');
+            window.uetq[5].should.eql(expectedUpdatedConsentPayload[2]);
+
+            done();
+        });
+
+        it('should ignore Unspecified Consent Settings if NOT explicitly defined in Consent State', function(done) {
+            mParticle.forwarder.init(
+                {
+                    tagId: 'tagId',
+                    consentMappingWeb: JSON.stringify(consentMap),
+                    defaultAdStorageConsentWeb: 'Unspecified', // Should be overridden by user consent state
+                },
+                reportService.cb,
+                false // Disable testMode so we can test init
+            );
+
+            var expectedConsentPayload = [
+                'consent',
+                'default',
+                { ad_storage: 'granted' },
+            ];
+
+            console.log('uetq', window.uetq);
+
+            window.uetq.length.should.eql(3);
+            window.uetq[0].should.equal('consent');
+            window.uetq[1].should.equal('default');
+            window.uetq[2].should.eql(expectedConsentPayload[2]);
```

Comment:
same comment as above about having uetq deep equal expectedConsentPayload.  I suppose I could see why you only use expectedConsentPayload[2], but then there isn't a need to have the full `expectedConsentPayload`, because you are retyping `consent` and `default` for the tests and not using the full `expectedConsentPayload`.